test/serializers/serialization_array_test.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  class Serializer
+    class SerializationArrayTest < ActiveSupport::TestCase
+      class ArraySerializer < ActiveModel::Serializer
+        attributes :total, :items
+
+        def total
+          object.count
+        end
+
+        def items
+          CollectionSerializer.new(object).as_json
+        end
+      end
+
+      class Author < ActiveModelSerializers::Model
+        attributes :id, :name
+      end
+      class AuthorSerializer < ActiveModel::Serializer
+        attributes :id, :name
+      end
+
+      setup do
+        @authors = [Author.new(id: 1, name: 'Blog Author')]
+        @serializer_instance = ArraySerializer.new(@authors)
+        @serializable = ActiveModelSerializers::SerializableResource.new(@authors, serializer: ArraySerializer, adapter: :attributes)
```

Comment:
https://github.com/rails-api/active_model_serializers/pull/2223 introduces the `fields` option for the `:attributes` adapter. So we can further trim down the fields that we want to render.
```ruby
@serializable = ActiveModelSerializers::SerializableResource.new(
  @authors,
  serializer: ArraySerializer,
  adapter: :attributes,
  fields: [:items]
)
```
But now this https://github.com/rails-api/active_model_serializers/blob/d182e60898d0fad956afbed4d9e0a5824baef06d/lib/active_model/serializer.rb#L369 will not be enough to prevent the issue:
> NoMethodError: undefined method `model_name'